mm2src/mm2_main/src/rpc/lp_commands/lp_commands_legacy.rs

```diff
@@ -37,45 +37,89 @@ use crate::mm2::lp_ordermatch::{cancel_orders_by, CancelBy};
 use crate::mm2::lp_swap::{active_swaps_using_coin, tx_helper_topic, watcher_topic};
 use crate::mm2::MmVersionResult;
 
+const INTERNAL_SERVER_ERROR_CODE: u16 = 500;
+const RESPONSE_OK_STATUS_CODE: u16 = 200;
+
 /// Attempts to disable the coin
 pub async fn disable_coin(ctx: MmArc, req: Json) -> Result<Response<Vec<u8>>, String> {
     let ticker = try_s!(req["coin"].as_str().ok_or("No 'coin' field")).to_owned();
-    let _coin = match lp_coinfind(&ctx, &ticker).await {
+    let coin = match lp_coinfind(&ctx, &ticker).await {
         Ok(Some(t)) => t,
         Ok(None) => return ERR!("No such coin: {}", ticker),
         Err(err) => return ERR!("!lp_coinfind({}): ", err),
     };
-    let swaps = try_s!(active_swaps_using_coin(&ctx, &ticker));
-    if !swaps.is_empty() {
-        let err = json!({
-            "error": format!("There're active swaps using {}", ticker),
-            "swaps": swaps,
-        });
-        return Response::builder()
-            .status(500)
-            .body(json::to_vec(&err).unwrap())
-            .map_err(|e| ERRL!("{}", e));
+    let coins_ctx = CoinsContext::from_ctx(&ctx).map_err(|err| ERRL!("{}", err))?;
+
+    // Get all enabled tokens with platform coin including the coin.
+    let coins_to_disable = coins_ctx
+        .get_tokens_to_disable(&ticker)
+        .await
+        .into_iter()
+        .chain(std::iter::once(coin.platform_ticker().to_string()));
+
+    // Check for matching order / active swaps
+    let mut cancelled_orders = vec![];
+    for ticker in coins_to_disable.clone() {
+        log!("disabling {ticker} coin");
+        let swaps = try_s!(active_swaps_using_coin(&ctx, &ticker));
+        if !swaps.is_empty() {
+            let err = json!({
+                "error": format!("There're active swaps using {}", &ticker),
+                "swaps": swaps,
+            });
+            return Response::builder()
+                .status(INTERNAL_SERVER_ERROR_CODE)
+                .body(json::to_vec(&err).unwrap())
+                .map_err(|e| ERRL!("{}", e));
+        }
+        let (cancelled, still_matching) = try_s!(
+            cancel_orders_by(&ctx, CancelBy::Coin {
+                ticker: ticker.to_string()
+            })
+            .await
+        );
+
+        if !still_matching.is_empty() {
+            let err = json!({
+                "error": format!("There're currently matching orders using {}", ticker),
+                "orders": {
+                    "matching": still_matching,
+                    "cancelled": &cancelled,
+                }
+            });
```

Comment:
What's your opinion @borngraced on not canceling any orders for all tokens if at least one of the tokens has one currently matching orders or active swaps?
In my opinion, when a user wants to disable a platform coin and all it's tokens, if at least one token still has an active swap or a matching order we shouldn't cancel orders for other tokens if they are not disabled. The alternative is to disable as much tokens as we can and return in the error the non disabled tokens because of active swaps/ matching orders which I think is also a good solution.
I would like your opinion too on this @sergeyboyko0791 :)

Reply:
> If I get you clear, for the first option, it's in the code already we'll return an error if there's a matching order for a token/coin which terminates the function from proceeding with disabling other tokens

What I meant in the first option is that right now some orders are cancelled without disabling any coins, this can be confusing for the user when calling the `disable_coin` for a platform coin and an error is returned while only cancelling some orders. It would be better to return an error without cancelling any orders in this case.

> for the second option I can do something like this to the error response returned if there is a matching order so it returns the token that are unable to be disabled
> ``` rust
>let err = json!({
>                "error": format!("There're currently matching orders using {}", ticker),
>                "orders": {
>                    "matching": still_matching,
>                    "cancelled": &cancelled,
>                    "not_disabled": coins_to_disable.filter(|c| !disabled_tokens.contains(c))
>                }
 >           });
>```

There might be more than 1 token that still matching orders or that has active swaps. All other tokens should be disabled for the second option, except the platform token since it should be disabled at the end only when all it's tokens are disabled. So you will need to do more changes than adding `not_disabled` field to the returned error when you find the first token with still matching orders.

Reply:
@shamardy I have just worked on this review and ready for another round of review thanks:)